src/retasc/product_pages_api.py

```diff
@@ -0,0 +1,99 @@
+
+    @cache
+    def active_releases(
+        self,
+        product_shortname: str,
+        fields: str = "shortname,product_shortname,ga_date,phase_display",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}?product__shortname={product_shortname}&phase__lt=Unsupported&fields=a,b,c
+        The response have the following fields:shortname,product_shortname,ga_date,phase_display
+        phase__lt=Unsupported means "any supported phase" (these are less than Unsupported).
+        """
+
+        opt = {
+            "product__shortname": product_shortname,
+            "phase__lt": "Unsupported",
+            "fields": fields,
+        }
+        res = self.session.get(self.api_url_releases(), params=opt)
+        res.raise_for_status()
+        return res.json()
+
+    @cache
+    def release_schedules(
+        self,
+        release_short_name: str,
+        fields: str = "name,date_start,date_finish,flags",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks?fields=a,b,c
+        The response have the following fields:name,date_start,date_finish,flags
+        """
+
+        url = self.api_url_release_schedule_tasks(release_short_name)
+        res = self.session.get(url, params={"fields": fields})
+        res.raise_for_status()
+        return res.json()
+
+    @cache
+    def release_schedules_by_search_query(
+        self, release_short_name: str, **kargs
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks/?{flag_cond}={flags}&{name__regex}={reg_query}&fields=a,b,c
+        """
+
+        url = self.api_url_release_schedule_tasks(release_short_name)
+        # fields has some default value, can be updated if kargs has it
+        addtional_query = {
+            "fields": "release_shortname,name,slug,date_start,date_finish"
+        }
+        for k, v in kargs.items():
+            addtional_query.update({k: v})
+
+        res = self.session.get(url, params=addtional_query)
+        res.raise_for_status()
+        return res.json()
+
+    @cache
+    def get_all_active_releases_tasks_for_product(self, product_shortname: str) -> dict:
+        """
+        Get all release names that are still supported by rest api:
+        https://{pp_release_url}?product__shortname={product_shortname}&phase__lt=Unsupported&fields=shortname
+        Loop through the releases and get the tasks of it by rest api
+        https://{pp_release_url}{release_short_name}/schedule-tasks/fields=name,slug,date_start,date_finish
+        """
+
+        releases_list = self.active_releases(product_shortname)
+        all_tasks: dict = {}
+        all_tasks[product_shortname] = {}
+        for r in releases_list:
+            r_name = r["shortname"]
+            r_tasks = self.release_schedules(r_name)
+            all_tasks[product_shortname][r_name] = r_tasks
+        return all_tasks
```

Comment:
This wraps the result unnecessarily in a dict containing just one item `product_shortname`.

```suggestion
        return {r["shortname"]: self.release_schedules(r["shortname"]) for r in releases_list}
```